packages/eslint-plugin/tests/rules/no-unsafe-declaration-merging.test.ts

```diff
@@ -80,23 +80,19 @@ class Foo {}
     },
     {
       code: `
-namespace Foo {
-  export interface Bar {}
-}
-namespace Foo {
-  export class Bar {}
-}
```

Comment:
note - we can no longer detect cases like this (as I mentioned here https://github.com/typescript-eslint/typescript-eslint/issues/5854#issuecomment-1288500991).

I think this is an okay trade-off as it should be really rare that someone is declaration merging via a declaration merged namespace.

Reply:
it seems good trade off. i tried to handle these kinds of cases but i couldn't finish until now